tools/clang/lib/Sema/SemaOverload.cpp

```diff
@@ -1922,12 +1922,14 @@ bool Sema::IsFloatingPointPromotion(QualType FromType, QualType ToType) {
       if (getLangOpts().HLSL) {
         if (FromBuiltin->getKind() == BuiltinType::LitFloat &&
             (ToBuiltin->getKind() == BuiltinType::Min10Float ||
+              ToBuiltin->getKind() == BuiltinType::HalfFloat ||
              ToBuiltin->getKind() == BuiltinType::Float ||
              ToBuiltin->getKind() == BuiltinType::Half ||
              ToBuiltin->getKind() == BuiltinType::Double))
           return true;
         if (FromBuiltin->getKind() == BuiltinType::Min10Float &&
             (ToBuiltin->getKind() == BuiltinType::Float ||
+             ToBuiltin->getKind() == BuiltinType::HalfFloat ||
```

Comment:
This area looks quite incomplete now, and I think it would benefit from using the `getFloatingRank` method in ASTContext.